compair/xapi/statement.py

```diff
@@ -48,6 +50,14 @@ def _add_default(cls, user, statement):
             referer_key = XAPIExtension.context_extensions.get('referer')
             statement.context.extensions[referer_key] = request.environ.get('HTTP_REFERER')
 
+        if impersonation.is_impersonating():
```

Comment:
instead, change `statement.actor = XAPIActor.generate_actor(user)` line 23 above to be:

```
        if impersonation.is_impersonating() and user.id == current_user.id:
            statement.actor = XAPIActor.generate_actor(impersonation.get_impersonation_original_user())
        else:
            statement.actor = XAPIActor.generate_actor(user)
```

and then here put:

```
        if impersonation.is_impersonating() and user.id == current_user.id:
            if not statement.context.extensions:
                statement.context.extensions = Extensions()
            impersonating_as_key = XAPIExtension.context_extensions.get('impersonating as')
            statement.context.extensions[impersonating_as_key] = XAPIActor.generate_actor(user)
```